ui/e2e/instances/details/index.spec.ts

```diff
@@ -250,3 +259,194 @@ test("the diagram on the instance page changes appearance dynamically", async ({
     "5px"
   );
 });
+
+test("the input/output panel responds to user interaction", async ({
+  page,
+}) => {
+  const newInstance = createInstance({
+    namespace,
+    path: simpleWorkflowName,
+  });
+
+  await expect(newInstance, "wait until process was completed").toBeDefined();
+  const instanceId = (await newInstance).instance;
+  await page.goto(`/${namespace}/instances/${instanceId}`);
+
+  const inputOutputPanel = page.getByTestId("inputOutputPanel");
+
+  await expect(
+    inputOutputPanel,
+    "It renders the input/output panel"
+  ).toBeVisible();
+
+  const copyButton = inputOutputPanel.locator("button").nth(0);
+  const resizeButton = inputOutputPanel.locator("button").nth(1);
+
+  const inputButton = inputOutputPanel
+    .getByRole("tablist")
+    .locator("button")
+    .nth(0);
+  const outputButton = inputOutputPanel
+    .getByRole("tablist")
+    .locator("button")
+    .nth(1);
+
+  const textarea = inputOutputPanel.locator(".view-lines");
+  const expectedInput = `{}`;
+  const expectedOutput = `{    "result": "Hello world!"}`;
+  const expectedOutputCopy = '{"result":"Hello world!"}';
+
+  await resizeButton.hover();
+  await expect(
+    page.getByText("maximize output"),
+    "it shows the correct text when hovering over the resize button"
+  ).toBeVisible();
+
+  const minimizedWidth = (await inputOutputPanel.boundingBox())?.width;
+
+  await resizeButton.click();
+
+  const maximizedWidth = (await inputOutputPanel.boundingBox())?.width;
+  if (minimizedWidth === undefined || maximizedWidth === undefined) {
+    throw new Error("could not get width of diagram panel");
+  }
+  expect(
+    maximizedWidth / minimizedWidth,
+    "The panel is significantly bigger after maximizing"
+  ).toBeGreaterThan(1.5);
+
+  await resizeButton.hover();
+  await expect(
+    page.getByText("minimize output"),
+    "it shows the correct text when hovering over the resize button"
+  ).toBeVisible();
+
+  await page.reload();
+
+  const currentWidthAfterReload = (await inputOutputPanel.boundingBox())?.width;
+  expect(
+    currentWidthAfterReload,
+    "after reloading the page, the panel is still maximized"
+  ).toEqual(maximizedWidth);
+
+  await resizeButton.click();
+  await inputButton.click();
+
+  await expect(textarea, "the text shows the expected input").toHaveText(
+    expectedInput
+  );
+
+  await outputButton.click();
+
+  await expect(textarea, "the text shows the expected output").toHaveText(
+    expectedOutput
+  );
+
+  // for writing to the clipboard and for reading from the clipboard we need to grant permissions here - if we are not allowing it universally in the playwright.config.ts
+  page.context().grantPermissions(["clipboard-write", "clipboard-read"]);
```

Comment:
There is a tiny issue here that we encounter when we first tried testing copy and paste functionalities. It does not work in every browser. But there is an easy fix:

you can delete this permission stuff here and add the following to the top of the test


```typescript
import { mockClipboardAPI } from "e2e/utils/testutils";

// [...]

test.beforeEach(async ({ page }) => {
  await mockClipboardAPI(page);
``` 

This will overwrite the browsers' clipboard API with a "fake clipboard API". So in the test, the copy and paste will not actually work, but we make sure that the browsers clipboard API was called. So we basically assume that copy and paste will work when the proper browser APIs have been called.